draft-lenders-dns-cbor.md

```diff
@@ -627,12 +555,100 @@ dns-response = [
+Text string sequences within any tables for shared items or argument items within the rump MUST not be added to _V_.
+If a sequence for which an array is already in _V_ is encountered, a shared item reference _i_ to that array in V replaces this sequence.
+This shared item reference _i_ means: take the sequence from the array at _V_\[_i_\] and put it into the surrounding array in place of _i_.
+The resulting rump should look like referencing the _i_-th string (depth first) in the sequence.
+
+The "application/dns+cbor" media type comes with an optional parameter "packed".
+If it is not provided, it is assumed to be 0.
+With packed=0, any CBOR object `obj` marked by the "application/dns+cbor" media type MUST explicitly be understood as `TBD28259(obj)`, unless it is already `obj` itself is already tagged explicitly with TBD28259 as a whole.
+
+### Example
+
+Take the following content (note that this is intentionally not legal "application/dns+cbor" to illustrate generality).
+
+~~~ edn
+[
+  "www", "example", "org",
+  ["svc", "www", "example", "org"],
+  "org", "example", "org", 42,
+  "svc", "www", "example", "org", 42
+]
+~~~
+{: #fig:name-compression-example-unpacked title="Unpacked example for implicit text string suffix sequence compression."}
+
+This would generate the following virtual table _V_.
+
+~~~ edn
+[
+    ["www", "example", "org"],
+    ["example", "org"],
+    ["org"],
+    ["svc", "www", "example", "org"],
+    ["org", "example", "org"]
+]
+~~~
+{: #fig:name-compression-example-table title="Implicit table of shared items for the example."}
+
+Note that the sequence "org", "example", "org" is added at index 4 with leading "org", instead of referencing index 2 + index 1, as it is its own distinct suffix sequence.
+
+The packed representation would thus be:
+
+~~~ edn
+TBD28259(
+  [
+    ["www", "example", "org"],
+    ["svc", simple(0)],
+    "org", simple(1), 42,
+    simple(3), 42
+  ]
+)
+~~~
+{: #fig:name-compression-example-packed title="The packed representation of the example."}
+
+Note, with "application/dns+cbor;packed=0" the surrounding TBD28259 can be elided (even though the content would not be parsable as application/dns+cbor).
+
+With, e.g., table setup tag 113, further packing can be achieved by nesting table packing them with TBD28259.
+
+~~~ edn
+TBD28259(
+  113(
+    [
+      ["org", 42],
+      [
+        ["www", "example", simple(5)],
+        ["svc", simple(0)],
+        simple(5), simple(1), sample(6),
+        simple(3), sample(6)
+      ]
+    ]
+  )
+)
+~~~
+{: #fig:name-compression-example-packed-113 title="The packed representation of the example with additional table setup."}
+
+Note, how the previous references in {{fig:name-compression-example-packed}} do not changed, even though the string `"org"` is prepended, as strings in tables are ignored when creating the implicit table.
```

Comment:
Could we get rid of this extra rule with by switching the tag order?

Reply:
Then the indices for name compression would be larger than the 113 ones... Which would increase the risk of names having larger references (but as they are more common in DNS messages they should be smaller).

Reply:
see above.